MedallionShell.Tests/Streams/MergedLinesEnumerableTestAsync.cs

```diff
@@ -0,0 +1,136 @@
+﻿#if NETCOREAPP2_2_OR_GREATER
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Medallion.Shell.Streams;
+using Moq;
+using NUnit.Framework;
+
+namespace Medallion.Shell.Tests.Streams
+{
+    public class MergedLinesEnumerableTestAsync
+    {
+        [Test]
+        public async Task TestOneIsEmpty()
+        {
+            var empty1 = new StringReader(string.Empty);
+            var nonEmpty1 = new StringReader("abc\r\ndef\r\nghi\r\njkl");
+
+            IAsyncEnumerable<string> asyncEnumerable1 = new MergedLinesEnumerable(empty1, nonEmpty1);
+            var list1 = await asyncEnumerable1.ToListAsync();
```

Comment:
- [ ] Is this okay? I wasn't sure if I should do this or do 

```cs
(await asyncEnumerable1.SequenceEqualAsync(ConvertToAsyncEnumerable(["abc", "def", "ghi", "jkl"])))
    .ShouldEqual(true, ...)
```
The downside of not materializing is that we need to materialize it again in the `ShouldEqual` method.

Reply:
This class doesn’t support multiple enumeration, so doing it the way you have it is preferred. 

If ToListasync is the one reason we’re bringing in systemlinqasync, we should just write that as a 2-line helper method in this class instead.

Reply:
@madelson `FuzzTest` in this file currently needs all of the following:
- `AsyncEnumerable.Range(int start, int count)`
- `Select<TSource,TResult> (this IAsyncEnumerable<TSource> source, Func<TSource,TResult> selector)`
- `ToArrayAsync<TSource> (this IAsyncEnumerable<TSource> source)`
Would you prefer adding shims for all of these over adding `System.Linq.Async`?

Reply:
Turns out only `ToListAsync` was needed.